src/Filter/StringFilter.php

```diff
@@ -29,13 +29,14 @@ class StringFilter extends Filter
 
     public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
     {
-        if (!$data || !\is_array($data) || !\array_key_exists('value', $data) || null === $data['value']) {
+        if (!$data || !\is_array($data) || !\array_key_exists('value', $data)) {
             return;
         }
 
-        $data['value'] = trim($data['value']);
+        $data['value'] = trim($data['value'] ?? '');
 
-        if (0 === \strlen($data['value'])) {
+        // default filter value
+        if (0 === \strlen($data['value']) && empty($data['type'])) {
```

Comment:
I feel like this check shouldn't even exists.
- If the SonataAdmin decided to apply the filter, I shouldn't check again.

Moreover, if you want to allow looking for empty value, you should also also looking for value with only spaces, like `       ` ; isn't it ?

Maybe a `trim` (true, false ? Or left, right, both, none ?) option should be added.

Reply:
I add ability to use spaces in filtering.